extensions/big-data-cluster/src/bigDataCluster/controller/clusterControllerApi.ts

```diff
@@ -150,102 +158,156 @@ export class ClusterController {
 			throw new ControllerError(error, localize('bdc.error.mountHdfs', "Error creating mount"));
 		}
 	}
 
-	public async refreshMount(mountPath: string): Promise<MountResponse> {
-		let auth = await this.authPromise;
-		const api = new DefaultApiWrapper(this.username, this.password, this._url, auth);
+	private async getMountStatusImpl(self: ClusterController, mountPath?: string): Promise<MountStatusResponse> {
+		const auth = await self.authPromise;
+		const api = new DefaultApiWrapper(self.username, self.password, self._url, auth);
 
+		const mountStatus = await api.listMounts('', '', mountPath);
+		return {
+			response: mountStatus.response,
+			mount: mountStatus.body ? JSON.parse(mountStatus.body) : undefined
+		};
+	}
+
+	public async refreshMount(mountPath: string, promptConnect: boolean = false): Promise<MountResponse> {
 		try {
-			const mountStatus = await api.refreshMount('', '', mountPath);
-			return {
-				response: mountStatus.response,
-				status: mountStatus.body
-			};
+			return await this.withConnectRetry<MountResponse>(this.refreshMountImpl, promptConnect, mountPath);
 		} catch (error) {
-			// TODO handle 401 by reauthenticating
 			throw new ControllerError(error, localize('bdc.error.refreshHdfs', "Error refreshing mount"));
 		}
 	}
 
-	public async deleteMount(mountPath: string): Promise<MountResponse> {
+	private async refreshMountImpl(self: ClusterController, mountPath: string): Promise<MountResponse> {
+		const auth = await self.authPromise;
+		const api = new DefaultApiWrapper(self.username, self.password, self._url, auth);
+
+		const mountStatus = await api.refreshMount('', '', mountPath);
+		return {
+			response: mountStatus.response,
+			status: mountStatus.body
+		};
+	}
+
+	public async deleteMount(mountPath: string, promptConnect: boolean = false): Promise<MountResponse> {
+		try {
+			return await this.withConnectRetry<MountResponse>(this.deleteMountImpl, promptConnect, mountPath);
+		} catch (error) {
+			throw new ControllerError(error, localize('bdc.error.deleteHdfs', "Error deleting mount"));
+		}
+	}
+
+	private async deleteMountImpl(mountPath: string): Promise<MountResponse> {
 		let auth = await this.authPromise;
 		const api = new DefaultApiWrapper(this.username, this.password, this._url, auth);
 
+		const mountStatus = await api.deleteMount('', '', mountPath);
+		return {
+			response: mountStatus.response,
+			status: mountStatus.body
+		};
+	}
+
+	/**
+	 * Helper function that wraps a function call in a try/catch and if promptConnect is true
+	 * will prompt the user to re-enter connection information and if that succeeds updates
+	 * this with the new information.
+	 * @param f The API function we're wrapping
+	 * @param promptConnect Whether to actually prompt for connection on failure
+	 * @param args The args to pass to the function
+	 */
+	private async withConnectRetry<T>(f: (...args: any[]) => Promise<T>, promptConnect: boolean, ...args: any[]): Promise<T> {
```

Comment:
Thoughts on this? I'm not completely happy with this but I couldn't think of a simpler way to handle it so any feedback here would be appreciated